### PR TITLE
NAS-105010 / 11.3 / fix(cloud sync): Take Snapshot option can not be used for datasets th…

### PR DESCRIPTION
**Pull request by themylogin**
…at have nested datasets

### CODE REVIEW

File: src/middlewared/middlewared/plugins/cloud_sync.py [hunk 1]
```diff
@@ -748,6 +748,10 @@ class Config:
                 verrors.add(f"{name}.snapshot", "This option can only be enabled for PUSH tasks")
             if data["transfer_mode"] == "MOVE":
                 verrors.add(f"{name}.snapshot", "This option can not be used for MOVE transfer mode")
+            if await self.middleware.call("pool.dataset.query",
+                                          [["name", "^", os.path.relpath(data["path"], "/mnt") + "/"],
+                                           ["type", "=", "FILESYSTEM"]]):
+                verrors.add(f"{name}.snapshot", "This option can not be used for datasets that have nested datasets")
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/cloud_sync.py]: "datasets that have nested datasets" looks odd at first glance...

lets get opinion from @freenas/docs-team

Comment (Mrt134) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: Couple of refinement ideas on either side of positive/negative logic:
"This option is only available for non-parent datasets."
"This option is not available for parent datasets."

Comment (themylogin) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: Can "parent dataset" really be used to say "a dataset that has children"?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: I think so, but I will leave it up to @freenas/docs-team / @Mrt134

Comment (william-gr) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: @freenas/docs-team  ping

Comment (Mrt134) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: "parent" dataset was the term that appeared the most when referring to similar directory structuring, so I think it'll be largely understood.

Comment (Mrt134) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: I bounced it back and forth with Jaron a bit, and it seems "parent/child" terminology might not be immediately understood. How about `This option is only available for datasets that have no further nesting.` ?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/cloud_sync.py]: @themylogin

**Pull request merged**